parsers/OPENNEM.py

```diff
@@ -256,156 +245,147 @@ def filter_dataset(ds: dict) -> bool:
         zone_key=zone_key,
         session=session,
         target_datetime=target_datetime,
         logger=logger,
     )
-    region = ZONE_KEY_TO_REGION.get(zone_key)
-    capacities = get_capacities(filtered_datasets, region) if region else pd.Series()
-
     # Drop interconnectors
     df = df.drop([x for x in df.columns if "->" in x], axis=1)
-
-    # Make sure charging is counted positively
-    # and discharging negetively
+    # Make sure charging is counted positively and discharging negatively
     if "BATTERY_DISCHARGING" in df.columns:
         df["BATTERY_DISCHARGING"] = df["BATTERY_DISCHARGING"] * -1
 
     logger.debug("Preparing final objects..")
-    objs = [
-        {
-            "datetime": arrow.get(dt.to_pydatetime()).datetime,
-            "production": {  # Unit is MW
-                "coal": sum_vector(row, OPENNEM_PRODUCTION_CATEGORIES["coal"]),
-                "gas": sum_vector(row, OPENNEM_PRODUCTION_CATEGORIES["gas"]),
-                "oil": sum_vector(row, OPENNEM_PRODUCTION_CATEGORIES["oil"]),
-                "hydro": sum_vector(row, OPENNEM_PRODUCTION_CATEGORIES["hydro"]),
-                "wind": sum_vector(row, OPENNEM_PRODUCTION_CATEGORIES["wind"]),
-                "biomass": sum_vector(row, OPENNEM_PRODUCTION_CATEGORIES["biomass"]),
-                # We here assume all rooftop solar is fed to the grid
-                # This assumption should be checked and we should here only report
-                # grid-level generation
-                "solar": sum_vector(row, OPENNEM_PRODUCTION_CATEGORIES["solar"]),
-            },
-            "storage": {
-                # opennem reports charging as negative, we here should report as positive
-                # Note: we made the sign switch before, so we can just sum them up
-                "battery": sum_vector(row, OPENNEM_STORAGE_CATEGORIES["battery"]),
-                # opennem reports pumping as positive, we here should report as positive
-                "hydro": sum_vector(row, OPENNEM_STORAGE_CATEGORIES["hydro"]),
-            },
-            "capacity": {
-                "coal": sum_vector(capacities, OPENNEM_PRODUCTION_CATEGORIES["coal"]),
-                "gas": sum_vector(capacities, OPENNEM_PRODUCTION_CATEGORIES["gas"]),
-                "oil": sum_vector(capacities, OPENNEM_PRODUCTION_CATEGORIES["oil"]),
-                "hydro": sum_vector(capacities, OPENNEM_PRODUCTION_CATEGORIES["hydro"]),
-                "wind": sum_vector(capacities, OPENNEM_PRODUCTION_CATEGORIES["wind"]),
-                "biomass": sum_vector(
-                    capacities, OPENNEM_PRODUCTION_CATEGORIES["biomass"]
-                ),
-                "solar": sum_vector(capacities, OPENNEM_PRODUCTION_CATEGORIES["solar"]),
-                "hydro storage": capacities.get(OPENNEM_STORAGE_CATEGORIES["hydro"][0]),
-                "battery storage": capacities.get(
-                    OPENNEM_STORAGE_CATEGORIES["battery"][0]
-                ),
```

Comment:
question: @VIKTORVAV99 what is the recommended way to deal with this capacity data given that it cannot be captured in a `ProductionMix`? 

I see references to `OPENNEM.fetch_production_capacity` in some configs but I can't seem to find it. Is that a function that I need to write or was it dynamically generated from something else?

Reply:
So this is a small mess right now and I don't really know why but the capacity parsers live in an entirely different folder.

And yes as you mention there is no way to currently return the live capacity data. Honestly I'd leave it as a draft for now until I have discussed with the team on how to handle this.

But I'll probably end up creating a capacity mix class or something similar so we can keep parsing this data.

Reply:
Perfect! Thanks

Reply:
So I have been trying to figure out to do this, what I thought was going to be pretty straight forward is not as the space in the "hydro storage" and "battery storage" modes. This means they can't be used as variables which is kind of needed for it to plug into the existing solution we have. All workarounds I have create more errors than I can count 😅

There is a TODO about removing that space and simplifying things a bit and I think I'll have to do that first before working on the "capacity mixes".

Reply:
No worries!! Thanks for the update